client/finality-grandpa/src/finality_proof.rs

```diff
@@ -332,66 +298,78 @@ pub(crate) mod tests {
 		)
 	}
 
-	fn test_blockchain() -> InMemoryBlockchain<Block> {
-		use sp_finality_grandpa::GRANDPA_ENGINE_ID as ID;
-		let blockchain = InMemoryBlockchain::<Block>::new();
-		let just0 = Some(Justifications::from((ID, vec![0])));
-		let just1 = Some(Justifications::from((ID, vec![1])));
-		let just2 = None;
-		let just3 = Some(Justifications::from((ID, vec![3])));
-		blockchain.insert(header(0).hash(), header(0), just0, None, NewBlockState::Final).unwrap();
-		blockchain.insert(header(1).hash(), header(1), just1, None, NewBlockState::Final).unwrap();
-		blockchain.insert(header(2).hash(), header(2), just2, None, NewBlockState::Best).unwrap();
-		blockchain.insert(header(3).hash(), header(3), just3, None, NewBlockState::Final).unwrap();
-		blockchain
+	fn test_blockchain(
+		number_of_blocks: u64,
+		to_finalize: &[u64],
+	) -> (Arc<TestClient>, Arc<TestBackend>, Vec<Block>) {
+		let builder = TestClientBuilder::new();
+		let backend = builder.backend();
+		let mut client = Arc::new(builder.build());
+
+		let mut blocks = Vec::new();
+		for _ in 0..number_of_blocks {
+			let block = client.new_block(Default::default()).unwrap().build().unwrap().block;
+			block_on(client.import(BlockOrigin::Own, block.clone())).unwrap();
+			blocks.push(block);
+		}
+
+		for block in to_finalize {
+			let just = block.encode();
+			client.finalize_block(BlockId::Number(*block), Some((ID, just))).unwrap();
```

Comment:
The justification we're creating here is actually just a block number. This probably made sense before the latest changes when the test code was generic on the justification type.

Reply:
It's just some `Vec<u8>`, but wanted something unique per block in case the tests checked for those justifications. I'll change this to something simpler